modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/ObjectEntryHelper.java

```diff
@@ -110,12 +117,145 @@ public ObjectEntry getObjectEntry(
 		return objectEntries.get(0);
 	}
 
+	public List<Map<String, Object>> getSchemaEntities(
+			long companyId, APIApplication.Endpoint endpoint)
+		throws Exception {
+
+		APIApplication.Schema responseSchema = endpoint.getResponseSchema();
+
+		ObjectDefinition schemaMainObjectDefinition =
+			_getSchemaMainObjectDefinition(companyId, responseSchema);
+
+		Map<APIApplication.Property, ObjectField> propertyObjectFieldHashMap =
+			_getAPIApplicationPropertyObjectFieldMap(
+				companyId, responseSchema, schemaMainObjectDefinition);
+
+		List<ObjectEntry> objectEntries = getObjectEntries(
+			companyId, null,
+			schemaMainObjectDefinition.getExternalReferenceCode());
+
+		List<Map<String, Object>> entities = new ArrayList<>();
+
+		for (ObjectEntry objectEntry : objectEntries) {
+			Map<String, Object> objectEntryProperties =
+				_getAllObjectEntryProperties(objectEntry);
+
+			Map<String, Object> entity = new HashMap<>();
+
+			for (Map.Entry<APIApplication.Property, ObjectField>
+					propertyObjectFieldEntry :
+						propertyObjectFieldHashMap.entrySet()) {
+
+				APIApplication.Property property =
+					propertyObjectFieldEntry.getKey();
+				ObjectField objectField = propertyObjectFieldEntry.getValue();
+
+				entity.put(
+					property.getName(),
+					objectEntryProperties.get(objectField.getName()));
```

Comment:
As you are using just the `objectField` name, maybe we could include a parameter in the `Property` interface, something like `getSourceFileName`:

```
	public interface Property {

		public String getDescription();

		public String getExternalReferenceCode();

		public String getName();

		public String getSourceFileName();

		public Type getType();

		public enum Type {

			AGGREGATION, ATTACHMENT, BOOLEAN, DATE, DATE_TIME, DECIMAL, INTEGER,
			LONG_INTEGER, LONG_TEXT, MULTISELECT_PICKLIST, PICKLIST,
			PRECISION_DECIMAL, RICH_TEXT, TEXT

		}

	}
```

and then parse it:

```
private List<APIApplication.Property> _getProperties(
	Map<String, Object> schemaProperties, long companyId) {

	return TransformUtil.transformToList(
		(ObjectEntry[])schemaProperties.get("apiSchemaToAPIProperties"),
		propertyObjectEntry -> {
			[...]

			ObjectField objectField =
				_objectFieldLocalService.getObjectField(
					(String)properties.get("objectFieldERC"),
					objectDefinition.getObjectDefinitionId());

			return new APIApplication.Property() {

				[...]

				@Override
				public String getSourceFieldName() {
					return objectField.getName();
				}

				[...]

			};
		});
}
```

Reply:
Done